### ISSUE TITLE
Generator does not clear errors after call to processModel

### ISSUE THREAD

**Issue by kerimoyle** [cellml/libcellml#442]
I'm trying to use the `Generator` in some user tutorials (with only one equation as there's still a [segfault issue](#432 ) with more than one :( ) and am getting some behaviour that's not consistent with how the `Validator` and `Parser` entities work.  Here's a slice from my tutorial:

```
    //  3.a Create a Generator instance and use it to process the model.  Output
    //      any errors to the terminal using the utility function printErrorsToTerminal
    //      called with your generator as argument.
    libcellml::Generator generator;
    generator.processModel(model);
    printErrorsToTerminal(generator);
```

This outputs the error:
`Variable 'x' in component 'distance_finder' of model 'tutorial_3_model' is used in an ODE, but it is not initialised.` ... which is entirely expected as being the point of this section of the tut.

Then we continue and initialise the variable:
```
    //  3.b Set the initial conditions of the distance variable such that x(t=0)=5
    distance->setInitialValue(5.0);
    printModelToTerminal(model);  // This confirms that the initial value has been set ... but ... 
    generator.processModel(model);
    printErrorsToTerminal(generator); // ... this still returns the same error as before. 
```

If I simply move the `distance->setInitialValue(5.0);` before the first call to the generator then it works fine, that is: the generator does not clear errors.  This is not consistent with how the validator and parser store their errors - @agarny please can we make it so that errors are cleared when `processModel` is called?

**Comment by agarny** [comment]
Hmm, good point! I agree that any Generator's errors should be cleared before processing the model (i.e. at the begninng of `Generator::GeneratorImpl::processModel()`. Ok, consider it done.

**Comment by kerimoyle** [comment]
Closed by #444
